packages/solid-query/src/createMutationState.ts

```diff
@@ -0,0 +1,59 @@
+import { type Accessor, createMemo, onCleanup } from "solid-js";
+import {
+    type DefaultError,
+    type Mutation,
+    type MutationCache,
+    type MutationFilters,
+    type MutationState,
+    type QueryClient,
+    replaceEqualDeep,
+    useQueryClient,
+} from "."
+
+type MutationStateOptions<TResult = MutationState> = {
+    filters?: MutationFilters
+    select?: (
+        mutation: Mutation<unknown, DefaultError, unknown, unknown>,
+    ) => TResult
+};
+
+function getResult<TResult = MutationState>(
+    mutationCache: MutationCache,
+    options: MutationStateOptions<TResult>,
+): Array<TResult> {
+    return mutationCache
+        .findAll(options.filters)
+        .map(
+            (mutation): TResult =>
+                (options.select
+                    ? options.select(
+                        mutation as Mutation<unknown, DefaultError, unknown, unknown>,
+                    )
+                    : mutation.state) as TResult,
+        )
+}
+
+export function createMutationState<TResult = MutationState>(
+    options: Accessor<MutationStateOptions<TResult>> = () => ({}),
+    queryClient?: Accessor<QueryClient>,
+): Array<TResult> {
```

Comment:
This should return an `Accessor<TResult>`. I think this will also fix the failing test too. Since solid hooks are run once, We need to preserve it's reactivity by returning a signal instead